src/Infrastructure/Share/Query/Repository/ElasticRepository.php

```diff
@@ -7,19 +7,29 @@
 use Assert\Assertion;
 use Elasticsearch\Client;
 use Elasticsearch\ClientBuilder;
+use Psr\Log\LoggerInterface;
 
 abstract class ElasticRepository
 {
     private string $index;
 
     private Client $client;
 
-    public function __construct(array $config, string $index)
+    public function __construct(array $elasticConfig, ?LoggerInterface $esLogger = null)
```

Comment:
I'm really against optional dependencies. Just add it, will be needed and you can control the visibility via config.

Reply:
I removed `?` but let default to null in order to allow subclasses to disable logger (in case of sensitive data, if logger is enabled on prod environment...)